lib/library.rb

```diff
@@ -0,0 +1,18 @@
+class Library
+  attr_reader :name, :books, :authors
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+  end
+
+  def add_authors(author)
+    @authors << author
+    author_books = @authors.map do |author|
+      author.books
+    end
+    author_books.flatten
```

Comment:
we wouldn't need to rebuild the whole catalog of books from EVERY author, we only need to add this author's books to our existing catalog.